demos/trajectory_control.py

```diff
@@ -0,0 +1,1026 @@
+				# 			F[idx[direction][0]],
+				# 			F[idx[direction][1]],
+				# 			F[idx[direction][2]])
+
+
+				# # Plotting real-time values for the different terms of the total force
+				if self.is_grabbing:
+					position = self.Kp * (self.x_des - self.x_obs)
+					velocity = self.Kd * (self.v_des - self.v_obs)
+					global TIME
+
+					if self.plot_real_time:
+						plt.plot(TIME, F.data.tolist()[3][0], 'bo')
+						plt.plot(TIME, position.data.tolist()[3][0], 'y+')
+						plt.plot(TIME, velocity.data.tolist()[3][0], 'g+')
+						plt.plot(TIME, self.desired_external_force.data.tolist()[3][0], 'ro')
+						plt.pause(0.0001)
+
+					else:
+						self.plot_dict['Fo'].append(F.data.tolist()[3][0])
+						self.plot_dict['P'].append(position.data.tolist()[3][0])
+						self.plot_dict['V'].append(velocity.data.tolist()[3][0])
+						self.plot_dict['Fd'].append(self.desired_external_force.data.tolist()[3][0])
+
+					TIME += 1
+
+
+	def generate_plots(self):
+		print("Generating plots")
+		for time in range(len(self.plot_dict['Fo'])):
+			plt.plot(time, self.plot_dict['Fo'][time], 'bo')
+			plt.plot(time, self.plot_dict['P'][time], 'y+')
+			plt.plot(time, self.plot_dict['V'][time], 'g+')
+			plt.plot(time, self.plot_dict['Fd'][time], 'ro')
+			plt.pause(0.0001)
+			# plt.show()
+
+
+	
+
+
+	def get_desired_external_force(self):
+		# Calculating desired_external_force to apply to grasp the object
+		desired_external_force = np.matrix(np.zeros(9)).T
+
+		if self.change_force_dynamically and self.is_grabbing and self.t % 1000 == 0:
+			global TIME
+			# self.f_applied[0] = (1 - math.cos(TIME/10)) * 1.5
+			# self.f_applied[1] = (1 - math.cos(TIME/10)) * 1.5
+			self.f_applied[0] += 0.1
+			self.f_applied[1] += 0.1
+
+		# Force along x axis
+		# desired_external_force[0] = f_x
+		desired_external_force[3] = -self.f_applied[0]
+		desired_external_force[6] = self.f_applied[0]
+
+		# Force along y axis
+		desired_external_force[1] = -self.f_applied[1]
+		# desired_external_force[4] = self.f_applied[]
+		# desired_external_force[7] = self.f_applied[]
+
+		# Force along z axis
+		desired_external_force[2] = self.f_applied[2]
+		desired_external_force[5] = self.f_applied[2]
+		desired_external_force[8] = self.f_applied[2]
+
+		self.desired_external_force = desired_external_force
+
+
+
+
+	def move(self, desired_position, offset=None, timeout=5, exit_before_timeout=True):
+		''' 
+		Moves the arms to the desired final position
+		'''
+		if offset is not None:
+			self.object_size += offset
+
+		# Initial Values
+		i = 0
+		obj_start_pos = self.object_position
+
+		while True:
+			# Slowly changing the y-position for the end_effector
+			x = (1 - math.cos(math.radians(i/10))) * (desired_position[0] - obj_start_pos[0])/2 + obj_start_pos[0]
+			y = (1 - math.cos(math.radians(i/10))) * (desired_position[1] - obj_start_pos[1])/2 + obj_start_pos[1]
+			z = (1 - math.cos(math.radians(i/10))) * (desired_position[2] - obj_start_pos[2])/2 + obj_start_pos[2]
```

Comment:
I have some difficulty understanding this equation.  It is a sine wave, so as `i` increases, `[x,y,z]` first approaches the desired position but when reached, it will move away again.  Maybe I am missing something, how is it ensured that it stops when it reaches the goal, especially if `exit_before_timeout = False`?